pyterrier/transformer.py

```diff
@@ -54,8 +54,17 @@ class Transformer:
     """
         Base class for all transformers. Implements the various operators ``>>`` ``+`` ``*`` ``|`` ``&`` 
         as well as ``search()`` for executing a single query and ``compile()`` for rewriting complex pipelines into more simples ones.
+
+        Its expected that either ``.transform()`` or ``.transform_iter()`` be implemented by any class extending this - this rule
+        does not apply for indexers, which instead implement ``.index()``.
     """
 
+    def __init__(self, *args, **kwargs):
```

Comment:
I worry about adding a constructor since there wasn't one previously, and people probably haven't been consistent in calling it. (I know I haven't.)